### PR TITLE
adds nullIfEmpty extension for String

### PR DESCRIPTION
**Pull request by MajaTrzebiatowska**
* adds nullIfEmpty extension for String

### CODE REVIEW

File: kt-extensions/src/test/java/com/mindera/skeletoid/kt/extensions/utils/StringUnitTest.kt [hunk 1]
```diff
@@ -232,4 +232,16 @@ class StringUnitTest {
...
+    fun testNullIfEmptyNull() {
+        val text: String? = null
+        assertEquals(null, text.nullIfEmpty())
+    }
+
+    @Test
+    fun testNullIfEmptyEmpty() {
+        val text: String = ""
+        assertEquals(null, text.nullIfEmpty())
+    }
```
Comment (bruno-garces) [inline kt-extensions/src/test/java/com/mindera/skeletoid/kt/extensions/utils/StringUnitTest.kt:246]: should we add a test for when a string is neither null nor empty?

**Review by bruno-garces — CHANGES_REQUESTED** [review]

File: kt-extensions/src/test/java/com/mindera/skeletoid/kt/extensions/utils/StringUnitTest.kt (hunk 1 above)
Comment (MajaTrzebiatowska) [reply kt-extensions/src/test/java/com/mindera/skeletoid/kt/extensions/utils/StringUnitTest.kt:246]: done!

**Review by bruno-garces — APPROVED** [review]

**Review by pedro-belchior — APPROVED** [review]

File: kt-extensions/src/test/java/com/mindera/skeletoid/kt/extensions/utils/StringUnitTest.kt [hunk 2]
```diff
@@ -232,4 +232,22 @@ class StringUnitTest {
...
+
+    @Test
+    fun testNullIfEmpty() {
+        val text: String = "text"
+        assertEquals("text", text.nullIfEmpty())
+    }
```
Comment (Jmvcaseiro) [inline kt-extensions/src/test/java/com/mindera/skeletoid/kt/extensions/utils/StringUnitTest.kt:252]: Are the tests for nullIfBlank somewhere else? If so maybe we can move them here? If there aren't any maybe we can add them here (since they are very similar to these)

Comment (Jmvcaseiro) [reply kt-extensions/src/test/java/com/mindera/skeletoid/kt/extensions/utils/StringUnitTest.kt:252]: (I'll approve it since this isn't actually a problem with this PR and you can just create a task and resolve the conversation if you don't want to do it now)

**Review by Jmvcaseiro — APPROVED** [review]
